File: backend/learnlang/src/main/java/tr/com/english/learnlang/entity/words/Word.java
package tr.com.english.learnlang.entity.words;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import tr.com.english.learnlang.entity.category.Category;
import tr.com.english.learnlang.entity.sentence.Sentence;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Word {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String mean;
    private String description;
    @OneToMany(cascade = CascadeType.ALL)
    private List<Sentence> exampleSentence;
    @ManyToOne(cascade =CascadeType.ALL )
    private Category category;


}

File: frontend/app/src/main/java/com/eng/learnlang/core/domain/model/Word.kt
package com.eng.learnlang.core.domain.model

data class Word(
    val id : Int?=null,
    val name : String?=null,
    val mean : String?=null,
    val description : String?=null,
    val exampleSentence : List<Sentence> ? =null,
    val category: Category?=null,
    var verified: Boolean=false,
    var isWordInMyWordList : Boolean = false
)

